builder/ebs/builder_acc_test.go

```diff
@@ -76,6 +76,73 @@ func TestAccBuilder_EbsRegionCopy(t *testing.T) {
 	acctest.TestPlugin(t, testCase)
 }
 
+func TestAccBuilder_EbsRegionsCopyWithDeprecation(t *testing.T) {
+	amiName := fmt.Sprintf("packer-test-builder-region-copy-deprecate-acc-test-%d", time.Now().Unix())
+
+	amis := []amazon_acc.AMIHelper{
+		{
+			Region: "us-east-1",
+			Name:   amiName,
+		},
+		{
+			Region: "us-west-1",
+			Name:   amiName,
+		},
+	}
+
+	deprecationTime := time.Now().UTC().AddDate(0, 0, 1)
+	deprecationTimeStr := deprecationTime.Format(time.RFC3339)
+	testCase := &acctest.PluginTestCase{
+		Name:     "amazon-ebs_region_copy_with_deprecation_test",
+		Template: fmt.Sprintf(testBuilderAccRegionCopyDeprecated, deprecationTimeStr, amiName),
+		Teardown: func() error {
+			err := amis[0].CleanUpAmi()
+			if err != nil {
+				t.Logf("ami %s cleanup failed: %s", amis[0].Name, err)
+			}
+			err = amis[1].CleanUpAmi()
+			if err != nil {
+				t.Logf("ami %s cleanup failed: %s", amis[1].Name, err)
+			}
+			return nil
+		},
+		Check: func(buildCommand *exec.Cmd, logfile string) error {
+			if buildCommand.ProcessState != nil {
+				if buildCommand.ProcessState.ExitCode() != 0 {
+					return fmt.Errorf("Bad exit code. Logfile: %s", logfile)
+				}
+			}
+
+			err := checkRegionCopy(amiName, []string{"us-east-1", "us-west-1"})
+			if err != nil {
+				return err
+			}
+
+			var errors []error
```

Comment:
This is not a suggested changed I'm just sharing for your information we have a pkg for capturing multi-errors github.com/hashicorp/go-multierror that we use within Packer plugins for things like.

Reply:
Great suggestion, I had no idea this existed, I'll adapt my code to use it, thanks!